d2g.go

```diff
@@ -106,22 +106,34 @@ func artifacts(artifacts map[string]dockerworker.Artifact) []genericworker.Artif
 		gwArt := new(genericworker.Artifact)
 		defaults.SetDefaults(gwArt)
 
-		gwArt.Expires = artifacts[name].Expires
+		gwArt.Expires = dwPayload.Artifacts[name].Expires
 		gwArt.Name = name
 		gwArt.Path = "artifact" + strconv.Itoa(i)
-		gwArt.Type = artifacts[name].Type
+		gwArt.Type = dwPayload.Artifacts[name].Type
 
 		gwArtifacts[i] = *gwArt
 	}
+
+	if dwPayload.Features.DockerSave {
+		gwArt := new(genericworker.Artifact)
+		defaults.SetDefaults(gwArt)
+
+		gwArt.Name = "public/dockerImage.tar.gz"
+		gwArt.Path = "image.tar.gz"
+		gwArt.Type = "file"
+
+		gwArtifacts = append(gwArtifacts, *gwArt)
+	}
+
 	return gwArtifacts
 }
 
-func command(payload *dockerworker.DockerWorkerPayload, dwImage Image, gwArtifacts []genericworker.Artifact, gwWritableDirectoryCaches []genericworker.WritableDirectoryCache) ([][]string, error) {
+func command(dwPayload *dockerworker.DockerWorkerPayload, dwImage Image, gwArtifacts []genericworker.Artifact, gwWritableDirectoryCaches []genericworker.WritableDirectoryCache) ([][]string, error) {
```

Comment:
Renamed to `dwPayload` for consistency throughout the rest of the file and for readability sake. Reader can easily understand further down into the function that it's the docker worker payload that's being referenced.